File: app/src/main/java/com/liweiyap/narradir/avalon/AvalonCharacterName.java
package com.liweiyap.narradir.avalon;

/**
 * https://stackoverflow.com/questions/8157755/how-to-convert-enum-value-to-int
 */
public final class AvalonCharacterName
{
    private AvalonCharacterName(){}

    public static final int MERLIN = 0;
    public static final int PERCIVAL = 1;
    public static final int LOYAL0 = 2;
    public static final int LOYAL1 = 3;
    public static final int LOYAL2 = 4;
    public static final int LOYAL3 = 5;
    public static final int LOYAL4 = 6;
    public static final int LOYAL5 = 7;
    public static final int ASSASSIN = 8;
    public static final int MORGANA = 9;
    public static final int MORDRED = 10;
    public static final int OBERON = 11;
    public static final int MINION0 = 12;
    public static final int MINION1 = 13;
    public static final int MINION2 = 14;
    public static final int MINION3 = 15;

    public static int getNumberOfCharacters()
    {
        return 16;
    }
}

File: app/src/main/java/com/liweiyap/narradir/avalon/AvalonCharacterName.kt
package com.liweiyap.narradir.avalon

object AvalonCharacterName {
    const val MERLIN: Int = 0
    const val PERCIVAL: Int = 1
    const val LOYAL0: Int = 2
    const val LOYAL1: Int = 3
    const val LOYAL2: Int = 4
    const val LOYAL3: Int = 5
    const val LOYAL4: Int = 6
    const val LOYAL5: Int = 7
    const val ASSASSIN: Int = 8
    const val MORGANA: Int = 9
    const val MORDRED: Int = 10
    const val OBERON: Int = 11
    const val MINION0: Int = 12
    const val MINION1: Int = 13
    const val MINION2: Int = 14
    const val MINION3: Int = 15

    @JvmStatic
    fun getNumberOfCharacters(): Int {
        return 16
    }

    @JvmStatic
    fun getDefaultNumberOfGoodCharacters(): Int {
        return 3
    }

    @JvmStatic
    fun getDefaultNumberOfEvilCharacters(): Int {
        return 2
    }
}